### PR TITLE
array functions 27-06

### PR DESCRIPTION
**Pull request by maksof-kashif**
check it and merge it (27-06-2018)

### CODE REVIEW

File: common-functions.js [hunk 1]
```diff
@@ -74,4 +74,128 @@ function monthNameWiseFormate(b) {
 function dateFormat(c) {
 	var days=['','1st','2nd','3rd','4th','5th','6th','7th','8th','9th','10th','11th','12th','13th','14th','15th','16th','17th','18th','19th','20th','21st','22nd','23rd','24th','25th','26th','27th','28th','29th','30th','31st']
 	return days[c];
+}
+
+
+function checkArrayEmptyOrNot(array) {
+	if (array === undefined || array.length == 0||(array[0]=="" && array.length==1)) return "array is empty";
+	else return "array is not empty Array = "+array;}
```
Comment (maksof-ashar) [inline common-functions.js]: It should return only true/false. not strting text

File: common-functions.js [hunk 2]
```diff
@@ -74,4 +74,184 @@ function monthNameWiseFormate(b) {
...
+
+	function deleteArrayRec(array, delIndex) {
+	arr=[];
+	for(var i=0;i<array.length;i++){
+		if(array[i]==delIndex) continue;
```
Comment (maksof-ashar) [inline common-functions.js]: add space after if.

File: common-functions.js [hunk 3]
```diff
@@ -74,4 +74,184 @@ function monthNameWiseFormate(b) {
...
+		else {
```
Comment (maksof-ashar) [inline common-functions.js]: If there is one line of code, use it in single line instead without {}

File: common-functions.js [hunk 4]
```diff
@@ -74,4 +74,184 @@ function monthNameWiseFormate(b) {
...
+			arr.push(array[i]);
+		}
+	}
+	return arr;
+}
+
+function deleteMultipleRecFromArray(array, delIndex) {
+	for(var i=0; i<array.length; i++){
```
Comment (maksof-ashar) [inline common-functions.js]: add space after for and if.

File: common-functions.js [hunk 5]
```diff
@@ -74,4 +74,167 @@ function monthNameWiseFormate(b) {
...
 	var days=['','1st','2nd','3rd','4th','5th','6th','7th','8th','9th','10th','11th','12th','13th','14th','15th','16th','17th','18th','19th','20th','21st','22nd','23rd','24th','25th','26th','27th','28th','29th','30th','31st']
 	return days[c];
-}
\ No newline at end of file
+}
+
+
+function checkArrayEmptyOrNot(array) {
+	if (array === 'undefined' || array.length == 0||(array[0]=="")) return true;
+	else return false;}
```
Comment (maksof-ashar) [inline common-functions.js]: } < this should come in next line.

File: common-functions.js [hunk 6]
```diff
@@ -74,4 +74,167 @@ function monthNameWiseFormate(b) {
...
+
+	function deleteArrayRec(array, delIndex) {
+		arr=[];
+		for(var i=0;i<array.length;i++){
```
Comment (maksof-ashar) [inline common-functions.js]: you can simply use splice function here

File: common-functions.js [hunk 7]
```diff
@@ -74,4 +74,167 @@ function monthNameWiseFormate(b) {
...
+			if(array[i]==delIndex) continue;
+			else arr.push(array[i]);
+		}
+		return arr;
+	}
+
+	function deleteMultipleRecFromArray(array, delIndex) {
+		for(var i=0; i<array.length; i++){
+			for(var j=0;j<delIndex.length;j++){
```
Comment (maksof-ashar) [inline common-functions.js]: I don't think this function will work correctly, recheck again.
second add space after for, if, etc.

Comment (maksof-kashif) [reply common-functions.js]: theek kaam karraha hai ye

File: common-functions.js [hunk 8]
```diff
@@ -74,4 +74,159 @@ function monthNameWiseFormate(b) {
...
+			if(JSON.stringify(array1[i])==JSON.stringify(array2[j])) arr.push(array2[j]);
+		}
+	}
+	return arr;
+}
+
+function DelMultiRecByIndex(array,delIndex) {
+	arr=[];
+	for(var i=0; i<array.length; i++){
+		for(var j=0;j<delIndex.length;j++){
```
Comment (maksof-ashar) [inline common-functions.js]: arr = [];
it should be var arr = [];

**Pull request merged**